impl/src/main/java/org/ehcache/impl/serialization/CompactJavaSerializer.java

```diff
@@ -65,7 +66,19 @@
    * @see Serializer
    */
   public CompactJavaSerializer(ClassLoader loader) {
+    this(loader, new TransientStateRepository());
+  }
+
+  public CompactJavaSerializer(ClassLoader loader, StateRepository stateRepository) {
     this.loader = loader;
+    this.readLookup = stateRepository.getPersistentConcurrentMap("readLookup");
```

Comment:
+1 otherwise it will become messy when multiple disparate classes do use the state repository.